src/DynamoApplications/StartupUtils.cs

```diff
@@ -180,24 +190,68 @@ public static void PreloadShapeManager(ref string geometryFactoryPath, ref strin
-        ///which if we are running on mac os or *nix will use different logic then SandboxLookup 
+        ///which if we are running on mac os or *nix will use different logic then SandboxLookup
+        /// <paramref name="hostName"/> Dynamo variation identified by host.</param>
         /// </summary>
-        private static IUpdateManager InitializeUpdateManager()
+        private static IUpdateManager InitializeUpdateManager(string hostName)
         {
             var cfg = UpdateManagerConfiguration.GetSettings(new SandboxLookUp());
-            var um = new Dynamo.Updates.UpdateManager(cfg);
+            var um = new Dynamo.Updates.UpdateManager(cfg)
+            {
+                HostName = hostName
+            };
             Debug.Assert(cfg.DynamoLookUp != null);
             return um;
         }
 
-        
+        /// <summary>
+        /// Use this overload to construct a DynamoModel when the location of ASM to use is known and host name is known.
+        /// </summary>
+        /// <param name="CLImode">CLI mode starts the model in test mode and uses a seperate path resolver.</param>
+        /// <param name="asmPath">Path to directory containing geometry library binaries</param>
+        /// <param name="hostName">Dynamo variation identified by host.</param>
+        /// <returns></returns>
+        public static DynamoModel MakeModel(bool CLImode, string asmPath = "", string hostName ="")
+        {
+            //get sandbox executing location - this is where libG will be located.
+            var exePath = Assembly.GetExecutingAssembly().Location;
+            var rootFolder = Path.GetDirectoryName(exePath);
+            //defaults - these will fail.
+            var preloaderLocation = "libg_0_0_0";
+            var geometryFactoryPath = Path.Combine(preloaderLocation, DynamoShapeManager.Utilities.GeometryFactoryAssembly);
+
+            try
+            {
+                if (!Directory.Exists(asmPath))
+                {
+                    throw new FileNotFoundException($"{nameof(asmPath)}:{asmPath}");
+                }
+                Version asmBinariesVersion = DynamoShapeManager.Utilities.GetVersionFromPath(asmPath);
+
+                //get version of libG that matches the asm version that was supplied from geometryLibraryPath.
+                preloaderLocation = DynamoShapeManager.Utilities.GetLibGPreloaderLocation(asmBinariesVersion, rootFolder);
+                geometryFactoryPath = Path.Combine(preloaderLocation, DynamoShapeManager.Utilities.GeometryFactoryAssembly);
+
+                //load asm and libG.
+                DynamoShapeManager.Utilities.PreloadAsmFromPath(preloaderLocation, asmPath);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("A problem occured while trying to load ASM or LibG");
+                Console.WriteLine($"{e?.Message} : {e?.StackTrace}");
+            }
```

Comment:
Up to this line, this is just like the method below. Rather than copying it, it should be a shared private function between the two, that loads ASM/LibG.

Reply:
I can make that happen. Thanks